src/ConfigManager.cpp

```diff
@@ -16,68 +16,82 @@ bool DEBUG_MODE = false;
 // Setup and Loop
 //
 void ConfigManager::setup() {
-  char magic[MAGIC_LENGTH];
-  char ssid[SSID_LENGTH];
-  char password[PASSWORD_LENGTH];
+  this->mode = basic;
 
-  DebugPrintln(F("Reading saved configuration"));
+  if (!this->disabledWIFI) {
+    char magic[MAGIC_LENGTH];
+    char ssid[SSID_LENGTH];
+    char password[PASSWORD_LENGTH];
 
-  DebugPrint(F("MAC: "));
-  DebugPrintln(WiFi.macAddress());
+    DebugPrint(F("MAC: "));
+    DebugPrintln(WiFi.macAddress());
 
-  EEPROM.get(0, magic);
-  EEPROM.get(MAGIC_LENGTH, ssid);
-  DebugPrint(F("SSID: \""));
-  DebugPrint(ssid);
-  DebugPrintln(F("\""));
-  EEPROM.get(MAGIC_LENGTH + SSID_LENGTH, password);
-  readConfig();
+    DebugPrintln(F("Reading saved configuration"));
+    EEPROM.get(0, magic);
 
-  if (memcmp(magic, magicBytes, MAGIC_LENGTH) == 0) {
-    WiFi.begin(ssid, password[0] == '\0' ? NULL : password);
-    if (wifiConnected()) {
-      DebugPrint(F("Connected to "));
+    if (memcmp(magic, magicBytes, MAGIC_LENGTH) == 0) {
+      EEPROM.get(MAGIC_LENGTH, ssid);
+      DebugPrint(F("SSID: \""));
       DebugPrint(ssid);
-      DebugPrint(F(" with "));
-      DebugPrintln(WiFi.localIP());
+      DebugPrintln(F("\""));
+
+      EEPROM.get(MAGIC_LENGTH + SSID_LENGTH, password);
+      readConfig();
+
+      WiFi.begin(ssid, password[0] == '\0' ? NULL : password);
+
+      if (wifiConnected()) {
+        DebugPrint(F("Connected to "));
+        DebugPrint(ssid);
+        DebugPrint(F(" with "));
+        DebugPrintln(WiFi.localIP());
 
-      WiFi.mode(WIFI_STA);
-      startApi();
-      return;
+        WiFi.mode(WIFI_STA);
+      }
+    } else {
+      // We are at a cold start, don't bother timing out.
+      apTimeout = 0;
+      DebugPrintln(F("MagicBytes mismatch"));
     }
-  } else {
-    // We are at a cold start, don't bother timing out.
-    DebugPrint(F("MagicBytes mismatch - SSID: "));
-    DebugPrintln(ssid);
```

Comment:
attempting to output the `ssid` seemed off, as we got here because the data retrieval went sideways